tools/base/fpgainfo/main.c

```diff
@@ -0,0 +1,265 @@
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+#include <errno.h>
+#include <getopt.h>
+#include <stdlib.h>
+#include <string.h>
+#include <locale.h>
+#ifdef _WIN32
+#define EX_OK 0
+#define EX_USAGE (-1)
+#define EX_SOFTWARE (-2)
+#define EX_TEMPFAIL (-3)
+#else
+#include <sysexits.h>
+#endif
+
+#include "safe_string/safe_string.h"
+#include "argsfilter.h"
+#include "opae/fpga.h"
+
+//#include "sysinfo.h"
+#include "fpgainfo.h"
+
+#include "errors.h"
+#include "fmeinfo.h"
+#include "portinfo.h"
+#include "tempinfo.h"
+#include "powerinfo.h"
+#include "bmcinfo.h"
+
+void help(void);
+
+typedef fpga_result (*filter_fn)(fpga_properties *, int, char **);
+typedef fpga_result (*command_fn)(fpga_token *, int, int, char **);
+typedef void (*help_fn)(void);
+
+// define a list of command words and
+// function ptrs to the command handler
+static struct command_handler {
+	const char *command;
+	filter_fn filter;
+	command_fn run;
+	help_fn help;
+} cmd_array[] = {
+	{.command = "errors",
+	 .filter = errors_filter,
+	 .run = errors_command,
+	 .help = errors_help},
+	{.command = "power",
+	 .filter = power_filter,
+	 .run = power_command,
+	 .help = power_help},
+	{.command = "temp",
+	 .filter = temp_filter,
+	 .run = temp_command,
+	 .help = temp_help},
+	{.command = "fme",
+	 .filter = fme_filter,
+	 .run = fme_command,
+	 .help = fme_help},
+	{.command = "port",
+	 .filter = port_filter,
+	 .run = port_command,
+	 .help = port_help},
+	{.command = "perf",
+	 .filter = bmc_filter,
```

Comment:
Should this be perf_filter?

Reply:
There isn't a 'perf_filter'. We reuse 'bmc_filter'.